### PR TITLE
Moveculling

### PR DESCRIPTION
**Pull request by thomasflynn**
Moved culling outside of renderCamera().  Culling will now happen before the left eye is drawn.  Culling will only happen once per frame now.  Yay :).  Culling also now uses the center camera of the camera rig.

GearVRf-DCO-1.0-Signed-off-by: Tom Flynn 
tom.flynn@samsung.com

### CODE REVIEW

**Comment by thomasflynn** [comment]
any comments on this?

File: GVRf/Framework/src/org/gearvrf/GVRViewManager.java [hunk 1]
```diff
@@ -462,6 +463,11 @@ void onDrawEyeView(int eye, float fovDegrees) {
         if (!(mSensoredScene == null || !mMainScene.equals(mSensoredScene))) {
             GVRCameraRig mainCameraRig = mMainScene.getMainCameraRig();
 
+            if(eye == 0) {
+                GVRPerspectiveCamera centerCamera = mainCameraRig.getCenterCamera();
+                cull(mMainScene.getNative(), centerCamera.getNative(), mRenderBundle.getMaterialShaderManager().getNative());
+            }
+
```
Comment (dknaan) [inline GVRf/Framework/src/org/gearvrf/GVRViewManager.java]: I think it would be better to use a flag to indicate if cull was already calculated. In the past I saw several places in the code where the left eye was not called before the right eye, so we can't assume it.

In general. I would prefer to use LEFT_EYE, RIGHT_EYE, BOTH_EYES enums instead of 0,1
There is already an enum definition in GVRConsole which we have to move to somewhere else and use everywhere.

Comment (chenchao1407) [inline GVRf/Framework/src/org/gearvrf/GVRViewManager.java]: By general design, I am thinking onDrawEyeView, which will be called twice per frame, will handle rendering for each eye. And onDrawFrame is for pre-draw processing for a frame. I am just wondering if there is any reason that you put the culling here but not onDrawFrame?

File: GVRf/Framework/src/org/gearvrf/GVRViewManager.java [hunk 2]
```diff
@@ -462,6 +463,11 @@ void onDrawEyeView(int eye, float fovDegrees) {
         if (!(mSensoredScene == null || !mMainScene.equals(mSensoredScene))) {
             GVRCameraRig mainCameraRig = mMainScene.getMainCameraRig();
 
+            if(eye == 0) {
+                GVRPerspectiveCamera centerCamera = mainCameraRig.getCenterCamera();
+                cull(mMainScene.getNative(), centerCamera.getNative(), mRenderBundle.getMaterialShaderManager().getNative());
```
Comment (jason2kim) [inline GVRf/Framework/src/org/gearvrf/GVRViewManager.java]: isn't this called again in the else { // if ... statement below?

File: GVRf/Framework/src/org/gearvrf/GVRViewManager.java (hunk 1 above)
Comment (thomasflynn) [inline GVRf/Framework/src/org/gearvrf/GVRViewManager.java]: I actually did that initially, but was seeing weird things happening.  Objects would flicker in and out of existence right in front of me.  It was wierd.  Moved it to here and everything worked perfectly.  I agree this isn't the cleanest.  Let fight with it a little more and see if i can get it working from onDrawFrame.

Comment (chenchao1407) [inline GVRf/Framework/src/org/gearvrf/GVRViewManager.java]: OK. Maybe by the end of this week to see if you can handle this. After that if there is still a problem, we can merge it firstly and then figure out if we can move it to onDrawFrame with another pull request.

File: GVRf/Framework/src/org/gearvrf/GVRViewManager.java [hunk 3]
```diff
@@ -462,6 +463,11 @@ void onDrawEyeView(int eye, float fovDegrees) {
         if (!(mSensoredScene == null || !mMainScene.equals(mSensoredScene))) {
             GVRCameraRig mainCameraRig = mMainScene.getMainCameraRig();
 
+            if(eye == 0) {
```
Comment (jason2kim) [inline GVRf/Framework/src/org/gearvrf/GVRViewManager.java]: Isn’t this if(eye == 0) statement is redundant, as it will go to the else statement below? Eye will be either 0 or 1. Also, it assumes it is called 0 first.

**Comment by thomasflynn** [comment]
ok, moved it to onDrawFrame().  Seems to be working fine now.  Weird.  Please test before merging.

**Comment by thomasflynn** [comment]
ok, @dknaan?

**Comment by dknaan** [comment]
Code looks ok. I also tried to run some samples and everything looks fine.
